workspace/__ardulib__/SSD1306/SSD1306.cpp

```diff
@@ -0,0 +1,103 @@
+
+#include "SSD1306.h"
+
+SSD1306::SSD1306(TwoWire* wire, uint8_t i2cAddress, uint8_t width, uint8_t height) {
+    _wire = wire ? wire : &Wire;
+    _i2cAddress = i2cAddress;
+    _width = width;
+    _height = height;
+    _buffer = new uint8_t[_width * _height / 8];
```

Comment:
I have an idea, but I'm not sure that this is a popular case... Here it is:
- let's allocate memory outside this library and just accept the pointer to the buffer of the desired size (probably we need to check it here?)
- then we can allocate the buffer right in one of XOD nodes. E.G. `xod-dev/ssd1306/buffer`
- then this buffer might be reused by multiple displays of the same sizes (but only one should update it === call `render` method)
- and we don't need a deconstructor

Reply:
I agree with @brusherru for the point that the buffer pointer should be passed from outside, as an argument. But not for the reason mentioned (which is quite synthetic). If you’d define the buffer as a part of `State` in the node it will be stored in the data segment, not the heap. So, it will be clearly reported by the compiler as a thing which consumes RAM whereas in this case the pretty large consumption is masked.